README.md

```diff
@@ -243,26 +243,36 @@ class IterableWeakMap {
     set.delete(ref);
   }
 
-  constructor(iterable) {
-    for (const [key, value] of iterable) {
-      this.set(key, value);
+  constructor(iterable = null) {
+    if (iterable !== null) {
+      for (const { 0: key, 1: value } of iterable) {
```

Comment:
if leaks are a concern, for..of is not an available option. Perhaps `Array.from()`?

Reply:
This is the constructor, `Array.from` is no different when it comes to iterables.

---

Also, if it’s invoked as:
```js
new IterableWeakMap([[a, 1], [b, 2]])
```

then that’s no different from:
```js
new WeakMap([[a, 1], [b, 2]])
```
which also performs `for...of`‑like iteration

Reply:
On arraylikes, Array.from doesn’t invoke the iterator protocol, so it’s actually robust for the common case.

Reply:
@ljharb
`Array.from` prefers using the iterator protocol: <https://tc39.es/ecma262/#sec-array.from>

Reply:
ah, you’re right, i was thinking only of a missing Symbol.iterator i suppose.

in that case there’s no pragmatic way to use a robust iteration mechanism in the readme (altho https://npmjs.com/iterate-value does exist) so this can be resolved.